tests/util_support/test_date_disambiguator.py

```diff
@@ -81,3 +81,42 @@ def test_resolve_uncertain_int(self, date_str, min_or_max, expected):
     def test_assert_only_less_significant_uncertainty(self, date_str, expected_error):
         with pytest.raises(InvalidDate, match=expected_error):
             DateDisambiguator(date_str)
+
+    @freeze_time("2020-05-05")
+    @pytest.mark.parametrize(
+        "date_str, min_max_year, expected_range",
+        [
+            ("20XX-XX-XX", [2000, 2010], (datetime.date(2000, 1, 1), datetime.date(2010, 12, 31))),
+            ("2020-XX-XX", [2010, 2020], (datetime.date(2020, 1, 1), datetime.date(2020, 12, 31))),
+            ("2020-12-XX", [2010, 2020], (datetime.date(2020, 12, 1), datetime.date(2020, 12, 31))),
+
+            # The upper bound is in the future, which is valid.
+            ("2020-XX-XX", [2010, 2020], (datetime.date(2020, 1, 1), datetime.date(2020, 12, 31))),
+
+            # The upper bound defaults to the current date.
+            ("2020-05-XX", None, (datetime.date(2020, 5, 1), datetime.date(2020, 5, 5))),
+            ("20XX-XX-XX", [2000], (datetime.date(2000, 1, 1), datetime.date(2020, 5, 5))),
+        ],
+    )
+    def test_min_max_year(self, date_str, min_max_year, expected_range):
+        assert (
+            DateDisambiguator(date_str, min_max_year=min_max_year).range() == expected_range
+        )
+
+    @freeze_time("2020-05-05")
+    @pytest.mark.parametrize(
+        "date_str, min_max_year",
+        [
+            ("20XX-XX-XX", [1950, 1960]),
+            ("2020-XX-XX", [2000, 2010]),
+            ("2020-12-XX", [2000, 2010]),
+            ("2020-12-01", [2000, 2010]),
```

Comment:
This confused me. I thought 2020-12-01 would be exempt from these bounds since it contains no ambiguity.

Reply:
Oh, right. This test is misleading from the `augur refine --year-bounds` perspective. Un-ambiguous dates don't go through the `DateDisambiguator` class, but instead get parsed by `datetime.datetime.strptime`.

https://github.com/nextstrain/augur/blob/e7965fcb81c1a47d4c0c8d82e6e1641879ca5200/augur/dates.py#L115-L124

It's probably better to remove any non-ambiguous dates from this test file, since this class isn't ever used to parse those dates (though nothing is stopping it from doing so).

Reply:
This PR has been updated so that all dates with exact years are exempt from `min_max_year` bounds.